packages/wast-parser/src/validator.js

```diff
@@ -0,0 +1,47 @@
+// @flow
+
+import { traverse } from "@webassemblyjs/ast";
+
+// https://webassembly.github.io/spec/core/text/modules.html#text-module
+function atMostOneStartFunction(ast: Program) {
+  let startFuncs = 0;
+
+  traverse(ast, {
+    Start() {
+      startFuncs++;
+    }
+  });
+
+  if (startFuncs > 1) {
+    throw new Error("wat modules can only contain one Start function");
```

Comment:
For my own debugging purposes, I started using the `babel-code-frame` package which allows us to print the location in the code along with the error. In the future we could re-implement it with our own printer or with more information in the parser.